include/umf/memory_provider.h

```diff
@@ -17,6 +17,14 @@
 extern "C" {
 #endif
 
+/// @brief USM memory allocation type
+typedef enum umf_usm_memory_type_t {
+    UMF_MEMORY_TYPE_UNKNOWN = 0, ///< The memory pointed to is of unknown type
```

Comment:
I do not think we should put this enum in the `memory_provider.h` header. It is a USM-specific thing.
I understand your intention to share this enum between L0 and CUDA memory provider implementations. But it should be somewhere else, e.g. some header file that is included by `provider_cuda.h` and `provider_level_zero.h` headers.